config/routes.rb

```diff
@@ -304,10 +297,91 @@
   get "signup" => "static#show", id: "signup", constraints: { format: /(json|html)/ }
   get "login-preferences" => "static#show", id: "login", constraints: { format: /(json|html)/ }
 
+  get "my/*path", to: 'users#my_redirect'
+  get "user_preferences" => "users#user_preferences_redirect"
+
+  # New /u/ routes
```

Comment:
I might be missing something here but I was able to get the loop method working 

https://github.com/tgxworld/discourse/commit/bcefc080ac50e35b0e3ebe9460b31f5cd6667df9

I'm not quite sure what the problem `put` or `post` routes would cause. However, if we have routes with the `as: :some_path` options, the loop wouldn't work but based on the routes that we're changing we won't have that problem